src/parser.js

```diff
@@ -404,60 +339,32 @@ function convert(context) {
 
       case 'Literal':
         if (node.value === 'this') {
-          return makeNode('This', node.locationData);
+          return mapLiteral(context, node);
         } else {
           let start = linesAndColumns.indexForLocation({ line: node.locationData.first_line, column: node.locationData.first_column });
           let end = linesAndColumns.indexForLocation({ line: node.locationData.last_line, column: node.locationData.last_column }) + 1;
-          let raw = source.slice(start, end);
 
           let tokens = context.sourceTokens;
           let startTokenIndex = tokens.indexOfTokenContainingSourceIndex(start);
           let startTokenType = tokens.tokenAtIndex(startTokenIndex).type;
           let endTokenIndex = tokens.indexOfTokenContainingSourceIndex(end - 1);
           let endTokenType = tokens.tokenAtIndex(endTokenIndex).type;
           if (startTokenType === SourceType.JS) {
-            return makeNode('JavaScript', node.locationData, { data: node.value });
+            return mapLiteral(context, node);
           } else if (startTokenType === SourceType.HEREGEXP_START && endTokenType === SourceType.HEREGEXP_END) {
-            let flags = raw.match(HEREGEX_PATTERN)[2];
-            return makeNode('Heregex', node.locationData, {
-              quasis: [
-                makeNode('Quasi', node.locationData, { data: node.value })
-              ],
-              expressions: [],
-              flags: ['g', 'i', 'm', 'y'].reduce((memo, flag) => {
-                memo[flag] = flags.indexOf(flag) >= 0;
-                return memo;
-              }, {})
-            });
+            return mapLiteral(context, node);
+          } else if (startTokenType === SourceType.NUMBER) {
+            return mapLiteral(context, node);
           }
 
           let literal = parseLiteral(node.value);
```

Comment:
Doesn't need to be in this change, but looks like parseLiteral can go away pretty soon, or at least be greatly simplified? Glad to see babylon used for that stuff instead.